File: swedisch-vocab-app/src/main/java/de/isah/vocabtrainer/DebugInfoActivity.java
package de.isah.vocabtrainer;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.widget.TextView;

import de.isah.vocabtrainer.dictionary.Dictionary;
import de.isah.vocabtrainer.dictionary.DictionaryCache;
import de.isah.vocabtrainer.dictionary.constants.FileConstants;
import de.isah.vocabtrainer.logging.SwedishVocabAppLogger;

/**
 *
 */
public class DebugInfoActivity extends VocabTrainerAppCompatActivity {

    private DebugInfoHandler handler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SwedishVocabAppLogger.log("on create", DebugInfoActivity.class, isDebug);

        setContentView(R.layout.activity_debug_info);

        Dictionary dictionary = DictionaryCache.getCachedDictionary();
        this.handler = new DebugInfoHandler();

        String text = getDebugInfo(dictionary);

        TextView textView = findViewById(R.id.textViewDebugInfo);
        textView.setText(text);
    }

    @NonNull
    private String getDebugInfo(Dictionary dictionary) {
        return this.handler.getDebugInfo(dictionary, getResources().getString(R.string.app_name), getResources().getString(R.string.app_version));
    }

}

File: swedisch-vocab-app/src/main/kotlin/de/isah/vocabtrainer/DebugInfoActivity.kt
package de.isah.vocabtrainer

import android.os.Bundle
import android.widget.TextView
import de.isah.vocabtrainer.dictionary.Dictionary
import de.isah.vocabtrainer.dictionary.DictionaryCache
import de.isah.vocabtrainer.logging.SwedishVocabAppLogger

class DebugInfoActivity : VocabTrainerAppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_debug_info)
        SwedishVocabAppLogger.log("on create", DebugInfoActivity::class.java)
        val text = getDebugInfo()
        val textView = findViewById<TextView>(R.id.textViewDebugInfo)
        textView.text = text
    }

    fun getDebugInfo(dictionary: Dictionary = DictionaryCache.getCachedDictionary(), handler: DebugInfoHandler = DebugInfoHandler()): String{
        return handler.getDebugInfo(dictionary, resources.getString(R.string.app_name), resources.getString(R.string.app_version))
    }
}